### DISCUSSION TITLE
Why do I see the text "MQTT LightEntity" in a name?

### DISCUSSION THREAD

**Discussion by fernandolcardoso** [Koenkk/zigbee2mqtt#24108] [General]
I have a switch that turns on a light ...

![image](https://github.com/user-attachments/assets/9cf9e989-61da-4a84-89d0-b4e311e4dfeb)

I applied adjustments to the "configuration.yaml" file to change from switch to light ...

![image](https://github.com/user-attachments/assets/b21c45cc-c02f-4b90-993c-555efe002a28)

Before the configuration file adjustments, the switch name was displayed correctly without "MQTT LightEntity"

![image](https://github.com/user-attachments/assets/409946b4-e06a-4465-a311-d519e79f5df7)

I would like to know if it is possible to define the entity name through the configuration file, without having to change anything in the Home Assistant MQTT settings.

![image](https://github.com/user-attachments/assets/deb9e724-02c9-49a2-b9a3-5d97d02ebabc)

**Comment by Ziron** [comment]
I have been digging into this myself for a while also...

The [Home Assistant documentation for mqtt discovery](https://www.home-assistant.io/integrations/light.mqtt#min_mireds) states that:
> **name** string (Optional, default: MQTT Light)
> The name of the light. Can be set to `null` if only the device name is relevant.

Although this only works if the discovery message explicitly contains `{"name": null}`.
The current implementation of zigbee2mqtt just omits the `name` key from the discovery message instead if the configuration contains `name: null`.

This must have changed and been broken in some recent update, since another project [autoSteve/acMqtt#45](https://github.com/autoSteve/acMqtt/issues/45#issuecomment-2119096601) references zigbee2mqtt as an example of a working implementation regarding the same issue.

**Comment by Ziron** [comment]
After a bit more testing, it seems that *not* setting `name: null` in the configuration causes zigbee2mqtt to actually include `{"name": null}` in the discovery message, which then gives the wanted effect of using the device name for the light entity.

**Reply by fernandolcardoso** [reply]
Hi Ziron,

Sorry for the delay in getting back to you!

I performed the procedure to remove 'name: null' from the settings and now it works as expected.

![image](https://github.com/user-attachments/assets/810b0a7d-6a6c-4356-a37c-a441346f7b0d)


Thank you for your feedback and help.
